### PR TITLE
add repository, image, and token apis; add credential id to azure connection response

### PR DESCRIPTION
**Pull request by d-g-town**

### CODE REVIEW

File: porter/v1/cluster_control_plane.proto [hunk 1]
```diff
@@ -38,7 +38,10 @@ message CreateAzureConnectionRequest {
   bytes service_principal_secret = 5;
 }
 
-message CreateAzureConnectionResponse {}
+message CreateAzureConnectionResponse {
```
Comment (stefanmcshane) [inline porter/v1/cluster_control_plane.proto:41]: From the perspective of an API consumer, does this mean that the response will be a connection pool for azure?

Comment (stefanmcshane) [reply porter/v1/cluster_control_plane.proto:41]: I dont see this being used in the service either

File: porter/v1/cluster_control_plane.proto [hunk 2]
```diff
@@ -165,4 +216,16 @@ service ClusterControlPlaneService {
   // AssumeRoleCredentials should be used vary sparingly, and ONLY for replacing AWS Integrations which have no workaround on the Porter API.
   // This endpoint returns temporary AWS credentials for a given AWS Account ID, and should not be expanded further to allow specifc role selection without being tied to a project and cluster
   rpc AssumeRoleCredentials(AssumeRoleCredentialsRequest) returns (AssumeRoleCredentialsResponse) {}
+
+  // TokenForRegistry returns a docker-compatible token for accessing a given registry
+  rpc TokenForRegistry(TokenForRegistryRequest) returns (TokenForRegistryResponse) {}
+
+  // DockerConfigFileForRegistry returns a docker-compatible token for accessing a given registry
+  rpc DockerConfigFileForRegistry(DockerConfigFileForRegistryRequest) returns (DockerConfigFileForRegistryResponse) {}
```
Comment (stefanmcshane) [inline porter/v1/cluster_control_plane.proto:224]: Is this a Docker config as in the literal `docker config ls` config file, or a daemon.json, or a config.json?

File: porter/v1/cluster_control_plane.proto (hunk 1 above)
Comment (d-g-town) [reply porter/v1/cluster_control_plane.proto:41]: It's really just saving the credentials, so I'll go ahead and rename it to `SaveAzureCredentials`

File: porter/v1/cluster_control_plane.proto (hunk 2 above)
Comment (d-g-town) [reply porter/v1/cluster_control_plane.proto:224]: It's a config.json file (docker calls it a ConfigFile in their go code)

**Review by stefanmcshane — APPROVED** [review]

**Pull request merged**